File: 899-binary-gap/binary-gap.java
class Solution {
    public int binaryGap(int n) {
        StringBuilder s = new StringBuilder();
        while(n > 0){
            s.append((char)('0' + (n & 1)));
            n >>= 1;
        }
        int prev = -1;
        int ans = 0;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '1'){
                if(prev != -1){
                    ans = Math.max(ans, i - prev);
                }
                prev = i;
            }
        }
        return ans;
    }
}

File: 899-binary-gap/binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        string s = "";
        while(n){
            s.push_back('0'+(n&1));
            n>>=1;
        }
        int prev = -1;
        n = s.size();
        int ans = 0;
        for(int i=0; i<n; i++){
            if(s[i]=='1'){
                if(prev!=-1){
                    ans = max(ans, i-prev);
                }
                prev=i;
            }
        }
        return ans;
    }
};